### ISSUE TITLE
Master branch links are broken for branches with a `/` in the branch name

### ISSUE THREAD

**Issue by simonkotwicz** [sindresorhus/refined-github#2049]
Master branch links are broken for branches with a `/` in the branch name.

eg. branch `sk/dev` links to `[repo]/tree/master/dev`

bug is here: https://github.com/sindresorhus/refined-github/blob/master/source/features/branch-buttons.tsx#L78 
and here: https://github.com/sindresorhus/refined-github/blob/master/source/libs/page-detect.ts#L84

**Comment by bfred-it** [comment]
Can you post a real link to that so I don't have to go hunting for one? 😅

**Comment by simonkotwicz** [comment]
https://github.com/simonkotwicz/refined-github/tree/branch/name

**Comment by bfred-it** [comment]
- master links can be fixed but it needs to get the branch name from the branch selector dropdown instead of just the URL
- shortened URLs can't be fixed because we can't tell `branch/file` apart from `branch/still-branch` just by looking at the URL: https://github.com/bfred-it/shorten-repo-url/issues/9

**Comment by simonkotwicz** [comment]
right I guess the only other option would be change the format of the shortened URLs to be:
`owner/repo:branch/still-branch/folder`
OR
`owner/repo/tree(or blob)/branch/still-branch/folder`

**Comment by simonkotwicz** [comment]
What do you think about changing:

https://github.com/sindresorhus/refined-github/blob/master/source/features/branch-buttons.tsx#L78 

TO

[master/source/features/branch-buttons.tsx#L78 ](https://github.com/sindresorhus/refined-github/blob/master/source/features/branch-buttons.tsx#L78 )

That way at least the shortened URLs always work.

**Comment by bfred-it** [comment]
> I think the special `branch` formatting inside links is a little strange as well

I don't: commits, branches, tags, are all just git references, so https://github.com/bfred-it/shorten-repo-url tries to treat them all the same.

[simonkotwicz/refined-github:branch/still-branch/.travis.yml#L6 ](https://github.com/simonkotwicz/refined-github/blob/branch/still-branch/.travis.yml#L6) is still hard to parse compared to:

https://github.com/simonkotwicz/refined-github/blob/branch/still-branch/.travis.yml#L6

At most I'd try:

[simonkotwicz/refined-github/`branch`/still-branch/.travis.yml#L6 ](https://github.com/simonkotwicz/refined-github/blob/branch/still-branch/.travis.yml#L6)

---

**If anyone wants to pick this up,** change [this line](https://github.com/sindresorhus/refined-github/blob/73bef2da95b640e7ed39767ec181edb081d465e1/source/features/branch-buttons.tsx#L78) to use the URL from the branch selector instead of `location.pathname`. This would fix the `master` button on https://github.com/simonkotwicz/refined-github/blob/branch/still-branch/.travis.yml#L6

**Comment by kevin940726** [comment]
@bfred-it I would love to help! Maybe I can find time doing it later today.
